src/shared/workspace.utils.ts

```diff
@@ -0,0 +1,46 @@
+import { DataSourceInstanceSettings } from "@grafana/data";
+import { BackendSrv } from "@grafana/runtime";
+import { Workspace } from "core/types";
+
+export class WorkspaceUtils {
+    private _workspacesCache: Promise<Map<string, Workspace>> | null = null;
+
+    private readonly queryWorkspacesUrl = `${this.instanceSettings.url}/niauth/v1/auth`;
+
+    constructor(
+        readonly instanceSettings: DataSourceInstanceSettings,
+        readonly backendSrv: BackendSrv
+    ) {
+        this.loadWorkspaces();
+    }
+
+    get workspacesCache(): Promise<Map<string, Workspace>> {
+        return this._workspacesCache ?? this.loadWorkspaces();
+    }
```

Comment:
Exposing an async resolved property named "cache" feels like a name that is exposing implementation details. It's an unexpected name.  Could we instead make an async method that resolves to workspaces? Something like `async getWorkspaces(): Promise<Map<string, Workspace>>`